jquery.treetable.js

```diff
@@ -5,630 +5,626 @@
-
-      this.row = row;
-      this.tree = tree;
-      this.settings = settings;
-
-      // TODO Ensure id/parentId is always a string (not int)
-      this.id = this.row.data(this.settings.nodeIdAttr);
-
-      // TODO Move this to a setParentId function?
-      parentId = this.row.data(this.settings.parentIdAttr);
-      if (parentId != null && parentId !== "") {
-        this.parentId = parentId;
-      }
-
-      this.treeCell = $(this.row.children(this.settings.columnElType)[this.settings.column]);
-      this.expander = $(this.settings.expanderTemplate);
-      this.indenter = $(this.settings.indenterTemplate);
-      this.cell = $(this.settings.cellTemplate);
-      this.children = [];
-      this.initialized = false;
-      this.treeCell.prepend(this.indenter);
-      this.treeCell.wrapInner(this.cell);
-    }
-
-    Node.prototype.addChild = function(child) {
-      return this.children.push(child);
-    };
+(function ($) {
+    var Node, Tree, methods;
 
-    Node.prototype.ancestors = function() {
-      var ancestors, node;
-      node = this;
-      ancestors = [];
-      while (node = node.parentNode()) {
-        ancestors.push(node);
-      }
-      return ancestors;
-    };
+    Node = (function () {
+        function Node(row, tree, settings) {
+            var parentId;
 
-    Node.prototype.collapse = function() {
-      if (this.collapsed()) {
-        return this;
-      }
+            this.row = row;
+            this.tree = tree;
+            this.settings = settings;
 
-      this.row.removeClass("expanded").addClass("collapsed");
+            // TODO Ensure id/parentId is always a string (not int)
+            this.id = this.row.data(this.settings.nodeIdAttr);
 
-      this._hideChildren();
-      this.expander.attr("title", this.settings.stringExpand);
+            // TODO Move this to a setParentId function?
+            parentId = this.row.data(this.settings.parentIdAttr);
+            if (parentId !== null && parentId !== "") {
+                this.parentId = parentId;
+            }
 
-      if (this.initialized && this.settings.onNodeCollapse != null) {
-        this.settings.onNodeCollapse.apply(this);
-      }
+            this.treeCell = $(this.row.children(this.settings.columnElType)[this.settings.column]);
+            this.expander = $(this.settings.expanderTemplate);
+            this.indenter = $(this.settings.indenterTemplate);
+            this.children = [];
+            this.initialized = false;
+            this.treeCell.prepend(this.indenter);
+        }
 
-      return this;
-    };
+        Node.prototype.addChild = function (child) {
+            return this.children.push(child);
+        };
 
-    Node.prototype.collapsed = function() {
-      return this.row.hasClass("collapsed");
-    };
+        Node.prototype.ancestors = function () {
+            var ancestors, node;
+            node = this;
+            ancestors = [];
+            while (node === node.parentNode()) {
```

Comment:
This is supposed to be an assignment, not a comparison. By assigning the `node` to `node.parentNode()` in each run of the loop it walks from the leaf node to the root node to find all the ancestors :)

Reply:
I see! Silly me.
But to be honest, it's slightly confusing and unusual to see while loop and assignments together! :D

Reply:
How about the following?

            while (node !== null || node !== undefined) {
                node = node.parentNode();
                ancestors.push(node);
            }